joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala

```diff
@@ -0,0 +1,143 @@
+package io.joern.go2cpg.passes.ast
+
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+import io.shiftleft.codepropertygraph.generated.{ControlStructureTypes, DispatchTypes, Operators}
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+
+class AstCreationForArraysTests extends GoCodeToCpgSuite {
+  "AST Creation for Array Initialization" should {
+    "be correct when a int array is initialized" in {
+      val cpg = code("""
+          |package main
+          |func main() {
+          |	a := [5]int{1,2}
+          |}
+          |""".stripMargin)
+
+      val List(assignmentCallNode, arrayInitializerCallNode) = cpg.method("main").ast.isCall.l
+      assignmentCallNode.name shouldBe Operators.assignment
+      assignmentCallNode.code shouldBe "a := [5]int{1,2}"
+
+      arrayInitializerCallNode.name shouldBe Operators.arrayInitializer
+      arrayInitializerCallNode.code shouldBe "[5]int{1,2}"
+      arrayInitializerCallNode.typeFullName shouldBe "int[]"
+
+      cpg.method("main").ast.isLiteral.l.size shouldBe 2
```

Comment:
Instead of using `.ast` could we try to check `astChildren` of some node to be precise about the structure we're expecting?